examples/lfric/full_example/main.x90

```diff
@@ -0,0 +1,99 @@
+    use function_space_mod,   only: function_space_type
+    use fs_continuity_mod,    only: W0, W1, W2, W2V, W2H
+    use field_mod,            only: field_type, field_proxy_type
+    use testkern_w0_mod,      only: testkern_w0_type
+    use constants_mod,        only: r_def, i_def
+
+    implicit none
+
+    type(global_mesh_type), target             :: global_mesh
+    type(global_mesh_type), pointer            :: global_mesh_ptr
+    type(partition_type)                       :: partition
+    type(mesh_type), target                    :: mesh
+    type(uniform_extrusion_type), target       :: extrusion
+    type(uniform_extrusion_type), pointer      :: extrusion_ptr
+    type(function_space_type), target          :: vector_space
+    type(function_space_type), pointer         :: vector_space_ptr
+    procedure (partitioner_interface), pointer :: partitioner_ptr
+    type(field_type)                           :: field1, field2
+    type(field_proxy_type)                     :: proxy
+    integer(kind=i_def)                        :: lfric_fs = W0   ! W0
+    integer(kind=i_def)                        :: element_order = 1
+    integer(kind=i_def)                        :: ndata_sz
+
+    ! Use the unit-testing constructor:
+    global_mesh = global_mesh_type()
+    global_mesh_ptr => global_mesh
+
+    partitioner_ptr => partitioner_planar
+    partition = partition_type(global_mesh_ptr, &
+                               partitioner_ptr, &
+                               1, 1,  &    ! # procs in x and y direction
+                               0,     &    ! max stencil size
+                               0,     &    ! local rank
+                               1)          ! number of proceses
+
+    ! Creating a vertical mesh of height 100 m with 5 layers
+    ! It is important to use 0.0_r_def, otherwise:
+    ! Error: Component ‘atmosphere_bottom’ at (1) is a PRIVATE component of ‘extrusion_type’
+    ! Because it tries to use the Fortran default constructor to assign individual components
+    extrusion = uniform_extrusion_type(0.0_r_def, 100.0_r_def, 5)
+    extrusion_ptr => extrusion
+    mesh = mesh_type(global_mesh_ptr, partition, extrusion_ptr)
+    write (*,*) "Mesh has", mesh%get_nlayers(), "layers."
+    ndata_sz = 1
+    vector_space = function_space_type( mesh,          &
+                                        element_order, &
+                                        lfric_fs,      &
+                                        ndata_sz)
+    vector_space_ptr => vector_space
+    call field1%initialise( vector_space = vector_space_ptr, name="field1" )
+    call field2%initialise( vector_space = vector_space_ptr, name="field2" )
+    call invoke( name = 'Initialise fields',        &
+                 setval_c( field1,     0.0_r_def ), &
+                 setval_c( field2,     1.0_r_def )  &
+                 )
+    call invoke( name = 'testkern_w0', testkern_w0_type(field1, field2) )
+    proxy = field1%get_proxy()
```

Comment:
Fair point (as I've stated before, I assumed this would be replaced with your example). So I've removed the proxy and added as suggested the log_minmax call (which fails). For testing(!) I added:
```fortran
   call store_comm(1)
```
in main.x90 (after using it from mpi_mod), which avoids any further changes to any other file (though explicitly calling a function from the MPI module is not good). Or we could just set ``comm_set`` in mpi_mod to .true., this should make all functions to work as expected (and would be justified since we don't support MPI and have already modified mpi_mod). But I did not commit any of these changes to avoid conflicts with your work.

Reply:
@hiker, no worries, I modified `log_minmax` in my branch to work so I'm happy to update this example there.